test/raft/distributed_test.cc

```diff
@@ -328,5 +330,205 @@ struct rpc : public raft::rpc {
     }
 };
 
+// TODO: introduce yields and delays; use logical clock steered from outside as well?
+template <typename State, State init_state>
+struct persistence : public raft::persistence {
+    snapshots_t<State>& _snapshots;
+
+    std::pair<raft::snapshot, State> _stored_snapshot;
+    std::pair<raft::term_t, raft::server_id> _stored_term_and_vote;
+
+
+    // Invariant: for each entry except the first, the raft index is equal to the raft index of the previous entry plus one.
+    raft::log_entries _stored_entries;
+
+    // Instances of `store_log_entries` waiting for holes to be filled before they store their entries;
```

Comment:
As we discussed earlier there cannot be holes unless your persistence module is preemptable. Raft will store the log in strict order.